### PR TITLE
[IGNORE] Add adapter for ArchitectureComponents

### PR DESCRIPTION
**Pull request by pakoito**
The current implementation will simply not work because the tests time out. map/switchMap is so slow the tailRecM test takes longer than Travis' timeout. Back-of-the-envelope estimates are around 1h to complete.

This PR also includes important fixes to MonadLaws, as several laws were not being run. EDIT: Separated fix to #476

### CODE REVIEW

File: kategory-test/src/main/kotlin/kategory/laws/MonadLaws.kt [hunk 1]
```diff
@@ -45,15 +47,17 @@ object MonadLaws {
...
+                val res = M.tailRecM(0, { i -> M.pure(if (i < iter) Left(i + 1) else Right(i)) })
+                res.equalUnderTheLaw(M.pure(iter), EQ)
+            }
 
-    inline fun <reified F> stackSafetyComprehensions(iterations: Int = 5000, M: Monad<F> = monad<F>(), EQ: Eq<HK<F, Int>>): Unit {
-        val res = stackSafeTestProgram(M, 0, iterations)
-        res.run(M).equalUnderTheLaw(M.pure(iterations), EQ)
-    }
+    inline fun <reified F> stackSafetyComprehensions(iterations: Int = 5000, M: Monad<F> = monad<F>(), EQ: Eq<HK<F, Int>>): Unit =
+            forFew(1, Gen.oneOf(listOf(iterations))) { iter ->
```
Comment (pakoito) [inline kategory-test/src/main/kotlin/kategory/laws/MonadLaws.kt]: *gasp*

File: kategory-test/src/main/kotlin/kategory/laws/MonadLaws.kt [hunk 2]
```diff
@@ -45,15 +47,17 @@ object MonadLaws {
                 M.flatMap(fa, { M.pure(f(it)) }).equalUnderTheLaw(M.map(fa, f), EQ)
             })
 
-    inline fun <reified F> stackSafety(iterations: Int = 5000, M: Monad<F> = monad<F>(), EQ: Eq<HK<F, Int>>): Unit {
-        val res = M.tailRecM(0, { i -> M.pure(if (i < iterations) Left(i + 1) else Right(i)) })
-        res.equalUnderTheLaw(M.pure(iterations), EQ)
-    }
+    inline fun <reified F> stackSafety(iterations: Int = 5000, M: Monad<F> = monad<F>(), EQ: Eq<HK<F, Int>>): Unit =
+            forFew(1, Gen.oneOf(listOf(iterations))) { iter ->
```
Comment (pakoito) [inline kategory-test/src/main/kotlin/kategory/laws/MonadLaws.kt]: *gasp*

File: kategory-test/src/main/kotlin/kategory/laws/MonadLaws.kt [hunk 3]
```diff
@@ -69,8 +73,8 @@ object MonadLaws {
                     val c = pure(b + 1).bind()
                     yields(c)
                 }.run(M)
-                aa.equalUnderTheLaw(bb, EQ)
-                aa.equalUnderTheLaw(M.pure(num + 2), EQ)
+                aa.equalUnderTheLaw(bb, EQ) &&
```
Comment (pakoito) [inline kategory-test/src/main/kotlin/kategory/laws/MonadLaws.kt]: *sigh*

**Comment by nomisRev** [comment]
This is still `kategory` packages. Close @pakoito ?

**Comment by pakoito** [comment]
Keep it around, please. It's so it's easy to link.

**Comment by JorgeCastilloPrz** [comment]
In an effort to clean the PR list we've decided to close all the ones that are older than 2 months and consider them obsolete except for very specific circumstances that we will evaluate per PR. After talking to @pakoito we're closing this one.

**Pull request closed without merging**